src/ui/MobileSettings/MobileSettings.jsx

```diff
@@ -41,24 +44,48 @@ export const MobileSettings = () => {
   }, []);
 
   useEffect(() => {
-    //TODO: next step is implementation behavior
-    const handleClickOutside = (event) => {
+    let touchStartTime;
+    let touchEndTime;
+
+    const handleTouchStart = () => {
+      touchStartTime = new Date().getTime();
+    };
+    const handleClickOutside = () => {
       if (containerRef.current && !containerRef.current.contains(event.target)) {
         setIsSettingsMenuOpen(false);
         setIs2DMenuOpen(false);
         setIsCursorMenuOpen(false);
+        setTimeout(() => {
+          if (!isMenuHidden) {
+            setIsSettingsMenuOpen(false);
+            setIs2DMenuOpen(false);
+            setIsCursorMenuOpen(false);
+            setIsMenuHidden(true);
+          } else {
+            setIsMenuHidden(false);
+          }
+        }, 1000);
       }
     };
 
-    const containerElement = containerRef.current;
-    containerElement.addEventListener('click', handleClickOutside);
+    const handleTouchEnd = () => {
```

Comment:
what did you think about change a bit logic of delay instead of `Date` caluclation use this approach:

`
const [pressing, setPressing] = useState(false);
const pressTimer = useRef();
` 

`
const startPress = () => {
    setPressing(true);
    // Start a timeout for 1 second
    pressTimer.current = setTimeout(() => {
      if (pressing) {
        onLongPress();
      }
    }, 1000);
  };

  const endPress = () => {
    setPressing(false);
    clearTimeout(pressTimer.current); // Clear the timeout if the tuch is released
  };
`